src/fstab-generator/fstab-generator.c

```diff
@@ -1450,7 +1450,7 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "root")) {
 
-                if (proc_cmdline_value_missing(key, value))
+                if (arg_root_what || proc_cmdline_value_missing(key, value))
```

Comment:
Fair enough. I will rewrite it in such a way that the new parameter takes precedence over `root=` but in case the new parameter is not used at all, the _last_ value of `root=` is used if it's included multiple times in the command line. But let's agree on what that new parameter looks like first.